development_workflow.md

```diff
@@ -0,0 +1,69 @@
+# Development / Contribution Guidelines
+
+## General Development Workflow in the Lab
+
+
+
+### Forking and Branching
+
+The lab's general workflow for contributing to a project involves:
```

Comment:
We should mention that PRs must not be merged into the "master" branch with build failures or lack unit tests.  

Another subject worth discussing is what happens if a developer is working on a large feature that must be broken into multiple parts/PRs  (to minimize the size of the PR that others must review).

In this case, if the developer wishes to work incrementally on a large feature, the project maintainer(s) might (1) create a branch on the main repository for that feature (2) have the developer fork from that branch (3) contributors can then issue multiple, smaller PRs to that feature branch.  The smaller PRs to the feature branch do not necessarily need to build successfully or be unit tested.  The goal of this workflow is to encourage smaller PRs, consistent reviews of PRs, and when the feature branch is fully implemented and tested it is simply merged with master.